src/js/BookReader/utils/classes.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * Exposes a function from one class (FromClass) to another (ToClass), in such a way
+ * that if the ToClass's method is overriden, the FromClass's method is also overriden.
+ * WARNING: This modifies FromClass' prototype! So FromClasses cannot be shared between
+ * different ToClasses.
+ * @template TFrom the class we'll be getting the method from
+ * @template TTo the class we'll be adding the method to
+ * @param {new () => TFrom} FromClass the class to get the method from
+ * @param {string} fromMethod the method's name in FromClass
+ * @param {function(TFrom): TTo} fromTransform how to get the TTo `this` to use when setting the method on TFrom
+ * @param {new () => TTo} ToClass the class to add the method to
+ * @param {string} toMethod the name of the method to add to TTo (likely will be equal to fromMethod)
+ * @param {function(TTo): TFrom} toTransform how to get the TFrom this to use when calling the new method
+ */
+export function exposeLinked(FromClass, fromMethod, fromTransform, ToClass, toMethod, toTransform) {
```

Comment:
q: how long do we want our args list to be?

Reply:
:D Haha, why could you be asking :P I think I'll call this a special circumstance; I don't think there's a way to write this function that's not awkward/complex, so will just have to hope future devs use it without digging into it too deeply; we should work toward ripping it out when we find a good alternative way to have the same effect.